src/publisher/utils.py

```diff
@@ -317,20 +315,127 @@ def validate(struct, schema_path):
 
     try:
         schema = load_schema(schema_path)
-        validator(struct, schema)
+        jsonschema.validate(struct, schema)
         return struct
 
     except ValueError as err:
-        # your json schema is broken
+        # json schema is broken
         #raise ValidationError("validation error: '%s' for: %s" % (err.message, struct))
         raise
 
     except ValidationError as err:
-        # your json is incorrect
-        #LOG.error("struct failed to validate against schema: %s" % err.message)
+        # json is incorrect
+        v = jsonschema.Draft4Validator(schema)
+        err.error_list = list(v.iter_errors(struct))
+        err.count = len(err.error_list)
+        err.message, err.trace = format_validation_error_list(err.error_list, schema_path)
+        raise err
+
+def flatten_validation_errors(error):
+    """an error can have sub-errors and each sub-error can have it's own errors (a tree).
+    two nodes at different depths may be equally important in determining failure.
+    this visits each node depth-first, returning a single list of errors that can be sorted."""
+    rt = []
+    for suberror in error.context:
+        res = flatten_validation_errors(suberror)
+        rt.extend(res)
+
+    # finally, add parent to bottom of list
+    rt.append(error)
+    return rt
+
+def validation_error_detail(err, schema_file):
```

Comment:
This generates a nice view of the error.

Reply:
I've attached examples in the PR description

Reply:
They were well received and saved me some time. 😃